src/pymor/algorithms/vkoga/vkoga.py

```diff
@@ -0,0 +1,358 @@
+            idx_in_X = int(np.argmin(np.linalg.norm(self.X_train - mu, axis=1)))
+
+        # update the residual
+        if self._V is None and self._z is None:
+            self.res = self.F_train
+        else:
+            self.res = self.res - self._V[:, -1:] @ self._z[-1:]
+
+        self._extend_incremental(mu, idx_in_X)
+
+    def _extend_incremental(self, mu, idx_in_X):
+        """Incrementally add a new center to the existing interpolant."""
+        # compute k(yi, mu) for all training points y_i
+        self._K_X_mu = self.kernel(self.X_train, mu)
+
+        # update coefficient
+        z_new = (self.res[idx_in_X] / np.sqrt(self._power2[idx_in_X])).reshape(1, self.m)
+        self._z = np.vstack([self._z, z_new]) if self._z is not None else z_new
+
+        # update Cholesky inverse, Newton basis and the power function values
+        self._update_newton_basis(idx_in_X)
+        self._update_cholesky_inverse(idx_in_X)
+        self._update_power_function_evals()
+
+        # update centers and final coefficients
+        self._coefficients = self._C.T @ self._z
+        if self._centers is None and self._centers_idx is None:
+            self._centers = np.atleast_2d(mu)
+            self._centers_idx = np.array([idx_in_X], dtype=int)
+        else:
+            self._centers = np.vstack([self._centers, mu])
+            self._centers_idx = np.concatenate([self._centers_idx, np.array([idx_in_X], dtype=int)])
+
+    def _update_newton_basis(self, idx_in_X):
+        """Append Xi as the last block to the Newton basis V."""
+        if self._V is None:
+            self._V = self._K_X_mu / np.sqrt(self._power2[idx_in_X])
+        else:
+            Xi = self._K_X_mu - (self._V[idx_in_X] @ self._V.T).reshape(-1, 1)
+            Xi = Xi / np.sqrt(self._power2[idx_in_X])
+            self._V = np.concatenate([self._V, Xi], axis=1)
+
+    def _update_cholesky_inverse(self, idx_in_X):
+        """Extend the inverse of the Cholesky matrix by the new block."""
+        c_nn = 1 / self._V[idx_in_X, -1]
+        if self._C is None:
+            self._C = np.atleast_2d(c_nn)
+        else:
+            self._C = np.block([[self._C, np.zeros((self._C.shape[0], 1))],
+                                [- c_nn * self._V[idx_in_X, :-1] @ self._C, c_nn]])
+
+    def _update_power_function_evals(self):
+        """Incrementally update self._power2 after adding new center."""
+        # update power2: p_{n+1}^2(x) = p_n^2(x) - norms
+        norms = self._V[:, -1] ** 2
+        self._power2 = np.maximum(self._power2 - norms, 0)
+
+
+class VKOGAEstimator(BasicObject):
+    """Scikit-learn-style estimator using the :class:`VKOGASurrogate`.
+
+    The estimator uses the :func:`~pymor.algorithms.greedy.weak_greedy` in its `fit`-method
+    to select centers according to the given criterion.
+
+    The algorithm is described in :cite:`WH13` and :cite:`SH21`.
+
+    Parameters
+    ----------
+    kernel
+        Kernel to use in the estimator.
```

Comment:
Add some documentation what data type is expected here. Also, does the kernel have to satisfy some property? (I remember there was some discussion about non-diagonal kernels.)